app/src/main/java/org/p2p/wallet/auth/gateway/repository/mapper/GatewayServiceErrorMapper.kt

```diff
@@ -48,4 +49,49 @@ class GatewayServiceErrorMapper {
             }
         }
             .also { Timber.tag("GatewayServiceMapper").i(error.toString()) }
+
+    fun fromNetwork(error: RpcResponse.Error): PushServiceError =
+        when (error.code) {
+            -32050 -> PushServiceError.TemporaryFailure(
```

Comment:
what do you think to write it in more declarative way?

something like this:
enum class PushServiceErrorCode(val clazz: KClass<Any>, val supportedCodes: Set<Int>) {
   TemporaryFailure(PushServiceError.TemporaryFailure::class, setOf(-32050)),
   PhoneNumberAlreadyConfirmed(PushServiceError.PhoneNumberAlreadyConfirmed::class, setOf(-32051));

   companion {
      fun createByCode(code: Int, message: String): PushServiceError {
         val error = values.filter { code in it.supportedCodes }.firstOrNull()
         return error.clazz.primaryConstructor?.call(code, message) ?: PushServiceError.UnknownFailure(code, message)
      }
   }
}